lib/src/sip_ua_helper.dart

```diff
@@ -146,55 +162,61 @@ class SIPUAHelper extends EventManager {
     });
 
     eventHandlers.on(EventProgress(), (EventProgress event) {
       logger.debug('call is in progress');
-      _handleCallState(
-          'progress', event.response, null, event.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.PROGRESS, originator: event.originator));
     });
 
     eventHandlers.on(EventFailed(), (EventFailed event) {
       logger.debug('call failed with cause: ' + (event.cause));
-      _handleCallState(
-          'failed', event.response, null, event.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.FAILED, originator: event.originator));
       _session = null;
     });
 
     eventHandlers.on(EventEnded(), (EventEnded e) {
       logger.debug('call ended with cause: ' + (e.cause));
-      _handleCallState('ended', null, null, e.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.ENDED, originator: e.originator));
       _session = null;
     });
     eventHandlers.on(EventCallAccepted(), (EventCallAccepted e) {
       logger.debug('call accepted');
-      _handleCallState('accepted', null, null, null, null, null);
+      _notifyCallStateListeners(CallState(CallStateEnum.ACCEPTED));
     });
     eventHandlers.on(EventConfirmed(), (EventConfirmed e) {
       logger.debug('call confirmed');
-      _handleCallState('confirmed', null, null, null, null, null);
+      _notifyCallStateListeners(CallState(CallStateEnum.CONFIRMED));
     });
     eventHandlers.on(EventHold(), (EventHold e) {
       logger.debug('call hold');
-      _handleCallState('hold', null, null, e.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.HOLD, originator: e.originator));
     });
     eventHandlers.on(EventUnhold(), (EventUnhold e) {
       logger.debug('call unhold');
-      _handleCallState('unhold', null, null, e.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.UNHOLD, originator: e.originator));
     });
     eventHandlers.on(EventMuted(), (EventMuted e) {
       logger.debug('call muted');
-      _handleCallState('muted', null, null, null, e.audio, e.video);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.MUTED, audio: e.audio, video: e.video));
     });
     eventHandlers.on(EventUnmuted(), (EventUnmuted e) {
       logger.debug('call unmuted');
-      _handleCallState('unmuted', null, null, null, e.audio, e.video);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.UNMUTED, audio: e.audio, video: e.video));
     });
     eventHandlers.on(EventStream(), (EventStream e) async {
       // Wating for callscreen ready.
       Timer(Duration(milliseconds: 100), () {
-        _handleCallState('stream', null, e.stream, e.originator, null, null);
+        _notifyCallStateListeners(
+            CallState(CallStateEnum.STREAM, stream: e.stream));
```

Comment:
Here, we need to pass the  `originator` parameter, you can know that `stream` is `local` or `remote` for proper display.

Reply:
adding originator